rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/HostsFileBasedInfrastructureManager.java

```diff
@@ -373,6 +366,56 @@ private void removeNodes(List<String> depNodeURLs) {
 
     }
 
+    @Override
+    protected void initializeRuntimeVariables() {
+        runtimeVariables.put(FREE_HOSTS_KEY, new HashMap<InetAddress, Integer>());
+        runtimeVariables.put(REGISTERED_NODES_KEY, new HashMap<String, InetAddress>());
+        runtimeVariables.put(REMOVED_HOSTS_KEY, new HashMap<InetAddress, Integer>());
+        runtimeVariables.put(PN_TIMEOUT_KEY, new HashMap<String, Boolean>());
+    }
+
+    private void atomicRemoveNode(final Node node) {
```

Comment:
This method is a bit too long, is it possible to refactor it into multiple short private methods with descriptive names?